core/src/main/java/hudson/util/ProcessKillingVeto.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2004-2009, Sun Microsystems, Inc., Kohsuke Kawaguchi
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.util;
+
+import hudson.ExtensionPoint;
+import hudson.util.ProcessTreeRemoting.IOSProcess;
+
+import java.util.Collections;
+import java.util.List;
+
+import jenkins.model.Jenkins;
+
+/**
+ * Allows extensions to veto killing processes. If at least one extension vetoes the killing of a process, it will
+ * not be killed. This can be useful to keep daemon processes alive. An example is mspdbsrv.exe used by Microsoft
+ * compilers.
+ * 
+ * See JENKINS-9104
+ * 
+ * @author Daniel Weber <daniel.weber.dev@gmail.com>
+ */
+public abstract class ProcessKillingVeto implements ExtensionPoint {
+	/**
+	 * @return All ProcessKillingVeto extensions currently registered. An empty list if Jenkins is not available, never null.
+	 */
+	public static List<ProcessKillingVeto> all() {
+		Jenkins jenkins = Jenkins.getInstance();
+		if ( null == jenkins ) return Collections.emptyList();

```

Comment:
Commonly in Jenkins we use another order of parameters in comparisons. I'm not sure if it's defined somewhere in code conventions